middleware.go

```diff
@@ -0,0 +1,110 @@
+package jwt
+
+import (
+	"github.com/Digital-Insight-Technologies-Ltd/gin-jwt-cognito/models"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-errors/errors"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+)
+
+// UserClaimsKey is the key for user claims in our context
+const UserClaimsKey = "userClaims"
+
+// CognitoClaimsMiddleware is a middleware for Gin that extracts user claims from
+// AWS Cognito.
+func CognitoClaimsMiddleware() gin.HandlerFunc {
+	// parser
+	jwtParser := new(jwt.Parser)
+
+	return func(c *gin.Context) {
```

Comment:
Implement context timeouts or cancellations to prevent resource exhaustion in scenarios where token parsing or user context construction might hang or take too long. [medium]